File: fastify/src/service/UserSeriesService.ts
import { Userseries } from '../entity/Userseries';

export interface iUserSeriesService {
  findByPageAndSizeAndFilterAndStatusAndOrder(userId: number, page: number, size: number, filter?: string, status?: number, order?: string, ascendingDirection?: boolean): Promise<[Userseries[], number]>;
  findOne(userId: number, seriesId: number): Promise<Userseries>;
  save(userId: number, entity: Userseries): Promise<Userseries>;
  update(userId: number, seriesId: number, entity: Userseries): Promise<boolean>;
  remove(userId: number, seriesId: number): Promise<number>;
}

File: spring/src/main/java/hu/marko/szakdolgozat/spring/service/UserSeriesService.java
package hu.marko.szakdolgozat.spring.service;

import hu.marko.szakdolgozat.spring.service.model.PageModel;
import hu.marko.szakdolgozat.spring.service.model.UserSeries;

public interface UserSeriesService {
  PageModel<UserSeries> findByPageAndSizeAndFilterAndStatusAndOrder(Long userId, Integer page, Integer size,
      String filter,
      Long status, String order, Boolean ascendingDirection);

  UserSeries findOne(Long userId, Long seriesId);

  UserSeries save(Long userId, UserSeries entity);

  Boolean update(Long userId, Long seriesId, UserSeries entity);

  Long remove(Long userId, Long seriesId);
}
